### DISCUSSION TITLE
Render as string, from JSX as raw string

### DISCUSSION THREAD

**Discussion by GuyoST** [preactjs/preact#3985] [Q&A]
Hello every one,

I'm new here and I found out preact, and this page : https://github.com/preactjs/preact-render-to-string to render a component as string.

If we start from the example on this page, with the box and the fox, my question is.
Let's suppose we have a JSX file that contains the component :

```jsx
// Classical components work
class Fox extends Component {
	render({ name }) {
		return <span class="fox">{name}</span>;
	}
}

// ... and so do pure functional components:
const Box = ({ type, children }) => (
	<div class={`box box-${type}`}>{children}</div>
);
```

Then how would it be possible to render a component of this file, for example, same as the example : 

```jsx
let html = render(
	<Box type="open">
		<Fox name="Finn" />
	</Box>
);
```

I tried to do something with html-react-parser to parse JSX code but then I don't know what to do with the parsed JSX...
I feel i'm wrong doing this...

But can someone help me about it ?

My goal with that : allow a user to type JSX on a webpage, and render a component dynamically (static component, no interaction).

Thank you for your help (and I hope i'm at the good place to ask this question :) )

**Comment by rschristian** [comment]
Do you mean that you don't know what to do with the result of `preact-render-to-string` once you have it?

One use is `dangerouslySetInnerHTML` -- please take note of the name, this is dangerous! You should sanitize input.

```jsx
const exampleHTMLString = '<h1>Hello World!</h1>';

// Produces `<div><h1>Hello World!</h1></div>`
const MyComponent = () => <div dangerouslySetInnerHTML={{ __html: exampleHTMLString }} />
```

[Here's a little extension to your example](https://esm.rschristian.dev/#Ly8gU2ltcGxlIFByZWFjdCBydHMgZXhhbXBsZSBieSBAcnNjaHJpc3RpYW4KLy8gLS0tLS0tLS0tLS0tLS0tLQoKaW1wb3J0IHsgaHRtbCwgcmVuZGVyLCBDb21wb25lbnQgfSBmcm9tICdodHRwczovL2Nkbi5za3lwYWNrLmRldi9odG0vcHJlYWN0L3N0YW5kYWxvbmUnOwppbXBvcnQgeyByZW5kZXJUb1N0cmluZyB9IGZyb20gJ2h0dHBzOi8vY2RuLnNreXBhY2suZGV2L3ByZWFjdC1yZW5kZXItdG8tc3RyaW5nJzsKCi8vIENsYXNzaWNhbCBjb21wb25lbnRzIHdvcmsKY2xhc3MgRm94IGV4dGVuZHMgQ29tcG9uZW50IHsKCXJlbmRlcih7IG5hbWUgfSkgewoJCXJldHVybiBodG1sYDxzcGFuIGNsYXNzPSJmb3giIHN0eWxlPSJjb2xvcjogd2hpdGUiPiR7bmFtZX08L3NwYW4+YDsKCX0KfQoKLy8gLi4uIGFuZCBzbyBkbyBwdXJlIGZ1bmN0aW9uYWwgY29tcG9uZW50czoKY29uc3QgQm94ID0gKHsgdHlwZSwgY2hpbGRyZW4gfSkgPT4gKAoJaHRtbGA8ZGl2IGNsYXNzPSR7YGJveCBib3gtJHt0eXBlfWB9PiR7Y2hpbGRyZW59PC9kaXY+YAopOwoKY29uc3QgcmVuZGVyZWRUb1N0cmluZyA9IHJlbmRlclRvU3RyaW5nKAogICAgaHRtbGA8JHtCb3h9IHR5cGU9Im9wZW4iPgogICAgICAgIDwke0ZveH0gbmFtZT0iRmlubiIgLz4KICAgIDwvJHtCb3h9PmAKKTsKCmNvbnN0IFdyYXBwZXIgPSAoKSA9PiBodG1sYDxkaXYgZGFuZ2Vyb3VzbHlTZXRJbm5lckhUTUw9JHt7IF9faHRtbDogcmVuZGVyZWRUb1N0cmluZyB9fSAvPmA7CgpyZW5kZXIoaHRtbGA8JHtXcmFwcGVyfSAvPmAsIGRvY3VtZW50LmJvZHkpOw==)
